tests/v2/integration/setup/main.go

```diff
@@ -2,116 +2,101 @@ package main
-	v1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
 	"github.com/rancher/rancher/tests/framework/extensions/token"
 	"github.com/rancher/rancher/tests/framework/pkg/config"
 	namegen "github.com/rancher/rancher/tests/framework/pkg/namegenerator"
-	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/rancher/rancher/tests/v2prov/clients"
+	"github.com/rancher/rancher/tests/v2prov/cluster"
+	testdefaults "github.com/rancher/rancher/tests/v2prov/defaults"
 	"github.com/sirupsen/logrus"
-	"k8s.io/client-go/util/retry"
-)
-
-var (
-	agentTag         = os.Getenv("AGENT_TAG")
-	masterAgentImage = "rancher/rancher-agent:" + agentTag
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 const (
-	k3dClusterNameBasename = "k3d-cluster"
+	clusterNameBaseName = "integration-test-cluster"
 )
 
-// setup for integration testing
+// main creates a test namespace and cluster for use in integration tests.
 func main() {
-	rancherConfig := new(rancherClient.Config)
-
-	user := &management.User{
-		Username: "admin",
-		Password: "admin",
-	}
+	logrus.Infof("Generating test config")
+	ipAddress, err := getOutboundIP()
+	handleFatalf(err, "Error getting outbound IP address: %v", err)
 
-	logrus.Infof("Generating test config...")
-	ipAddress := getOutboundIP()
 	hostURL := fmt.Sprintf("%s:8443", ipAddress.String())
-	token, err := token.GenerateUserToken(user, hostURL)
-	if err != nil {
-		logrus.Fatalf("error with generating admin token: %v", err)
-	}
-
-	clusterName := namegen.AppendRandomString(k3dClusterNameBasename)
+	userToken, err := token.GenerateUserToken(
+		&management.User{
+			Username: "admin",
+			Password: "admin",
+		},
+		hostURL,
+	)
+	handleFatalf(err, "Error with generating admin token: %v", err)
 
 	cleanup := true
-	rancherConfig.AdminToken = token.Token
-	rancherConfig.Host = hostURL
-	rancherConfig.Cleanup = &cleanup
-	rancherConfig.ClusterName = clusterName
-
-	if err := defaults.Set(rancherConfig); err != nil {
-		logrus.Fatalf("error with setting up config file: %v", err)
+	rancherConfig := rancherClient.Config{
+		AdminToken:  userToken.Token,
+		Host:        hostURL,
+		Cleanup:     &cleanup,
+		ClusterName: namegen.AppendRandomString(clusterNameBaseName),
 	}
 
-	config.WriteConfig(rancherClient.ConfigurationFileKey, rancherConfig)
-
-	logrus.Infof("Setting up K3D downstream cluster...")
-	testSession := session.NewSession()
-
-	client, err := rancherClient.NewClient("", testSession)
-	if err != nil {
-		logrus.Fatalf("error creating admin client: %v", err)
-	}
-
-	agentSetting := &v3.Setting{}
-
-	agentSettingResp, err := client.Steve.SteveType("management.cattle.io.setting").ByID("agent-image")
-	if err != nil {
-		logrus.Fatalf("error get agent-image setting: %v", err)
-	}
-
-	err = v1.ConvertToK8sType(agentSettingResp.JSONResp, agentSetting)
-	if err != nil {
-		logrus.Fatalf("error converting to k8s type: %v", err)
-	}
-
-	agentSetting.Value = masterAgentImage
```

Comment:
One of the things that was an advantage for using the k3d cluster is being able to use the latest version of rancher-agent. What version of rancher-agent does the cluster now run? Does it not matter?

Reply:
As far as I know, the system agent version used in tests will be taken from `package/Dockerfile` (see [here](https://github.com/rancher/rancher/pull/41771/files#diff-d7ad7c7666dae3e86d4d3a850f7c4df6dcde86454e9995001ea017e44a3d35d5R46)). This should just be the latest system agent version. The old code was just using the value from the envvar AGENT_TAG, which I think has the same result.

Reply:
It's not the same result, that's why here: https://github.com/rancher/rancher/blob/cbf32214807bdb2ba6795b61f6bb79c6138beb66/scripts/ci#L14 we're building the docker images before validating. This build is happening from the latest commit, hence why in the setup it was updating the `management.cattle.io.setting` to use the latest image. That way changes to the agent image can be tested within the same PR.

Reply:
@cmurphy weren't you mentioning a possible use case to write integration tests against the "latest" agent changes?

Reply:
Thanks for the feedback. I've updated the test setup so it now creates an additional in-cluster registry and copies test images (i.e. `rancher-agent:master-head`) to it before using those images in the downstream clusters. Moving forward (but in a separate PR), I'll also update the `provisioning-test` setup to do the same thing so we're testing against the current rancher agent in those tests as well.

Reply:
Sounds good, but from what I remember the locally created rancher-agent image has the commit as the tag, rather it just simply being `master-head` hence why I was doing this
```
var (
 	agentTag         = os.Getenv("AGENT_TAG")
 	masterAgentImage = "rancher/rancher-agent:" + agentTag
 )
```